backend/app/controllers/spree/admin/resource_controller.rb

```diff
@@ -191,10 +192,14 @@ def parent_data
 
   def parent
     if parent?
-      @parent ||= self.class.parent_data[:model_class]
-                    .includes(self.class.parent_data[:includes])
-                    .find_by(self.class.parent_data[:find_by] => params["#{parent_model_name}_id"])
-      instance_variable_set("@#{parent_model_name}", @parent)
+      begin
+        @parent ||= self.class.parent_data[:model_class]
+                        .includes(self.class.parent_data[:includes])
+                        .find_by!(self.class.parent_data[:find_by] => params["#{parent_model_name}_id"])
+        instance_variable_set("@#{parent_model_name}", @parent)
+      rescue ActiveRecord::RecordNotFound => e
```

Comment:
For consistency I'd also pull this rescue to the method level. If this method was more complex I'd want to localize the rescue, but I think it's fine to prefer avoiding the git churn. It should be obvious to the reader where the exception might come from.

Reply:
Will do, thanks for the explanation :+1: